test/metabase/analytics/prometheus_test.clj

```diff
@@ -172,5 +172,8 @@
       (prometheus/inc! :metabase-email/unknown-metric))) ; << Does not throw.
   (testing "inc is recorded for known metrics"
     (with-prometheus-system! [_ system]
-      (prometheus/inc! :metabase-email/messages)
-      (is (< 0 (-> system :registry :metabase-email/messages ops/read-value))))))
+      (let [read-value #(-> system :registry % ops/read-value)]
+        (doseq [metric [:metabase-email/messages :metabase-metrics/adjust-errors]]
```

Comment:
If it's necessary to test this for every metric we introduce, then this is not the best way to go about it: we should have a way to figure out automatically what has to be tested (like we do with serdes stuff). But I suppose we don't have to test every metric.

Reply:
Yeah this is probably not needed, I will roll this back. But this also reminds me that I should now add a test for `inc-and-throw!` (assuming @dpsutton does not object to adding that wrapper to the lib).

Reply:
The tricky part with `inc-and-throw!` is to make sure we don't count the same error several times. I mean the other pattern we could follow to have a single catch in `adjust` that increments and re-throws. (This was my initial instinct.)

Reply:
Sure, could be nice to have the metrics recorded in a single place in `adjust`. This also has the added benefit that if _any_ exception is thrown, we record the error, even if it wasn't thrown directly by the metrics middleware, which I think is what we want. I will make this change.